### PR TITLE
Allow empty list inputs in `group-by` and return empty record

### PR DESCRIPTION
**Pull request by 0scvr**
# Description

Changed `group-by` behavior to accept empty list as input and return an empty record instead of throwing an error. I also replaced `errors_if_input_empty()` test to reflect the new expected behavior.

See #10713 

# User-Facing Changes

# Tests + Formatting

# After Submitting

### CODE REVIEW

File: crates/nu-command/tests/commands/group_by.rs [hunk 1]
```diff
@@ -97,9 +97,9 @@ fn errors_if_column_not_found() {
 }
 
 #[test]
-fn errors_if_input_empty() {
-    let actual = nu!("group-by date");
-    assert!(actual.err.contains("expected table from pipeline"));
+fn group_by_on_empty_list_returns_empty_record() {
+    let actual = nu!("[] | group-by a");
+    assert!(actual.err.is_empty());
```
Comment (amtoine) [inline crates/nu-command/tests/commands/group_by.rs:102]: maybe the example is not very good, but you could pass the output of your command, here `group-by` to `to nuon` to get a raw string, and then check that in `assert` :+1: 

```suggestion
    let actual = nu!("[{a: "foo", b: 1}, {a: "bar", b: 2}, {a: "foo", c: 3}] | group-by a | to nuon");
    assert_equal!(actual.out, "{foo: [{a: foo, b: 1}, {a: foo, c: 3}], bar: [[a, b]; [bar, 2]]}");
```

you might also want to format this and possibly use the `pipeline` function to break the pipeline on multiple lines :relieved:

**Review by amtoine — COMMENTED** [review]
> I'm still learning Rust and new to this codebase. How can I improve my unit test to check the correct output ? 
>
love that you tell that upfront, very happy to help you :pray: :blush: 

> I tried `assert_eq!(actual.out, "{}");` but it checks the stdout of `group-by` command and not the return value of the group_by function.
>
i've left a review below to hopefully show you one way it's commonly done in the source base of Nushell :wink:

File: crates/nu-command/tests/commands/group_by.rs (hunk 1 above)
Comment (sholderbach) [reply crates/nu-command/tests/commands/group_by.rs]: Good call! Surprised about the recusrion in `nu!` due to maybe the line breaks.

Comment (amtoine) [reply crates/nu-command/tests/commands/group_by.rs]: happy to help @0scvr :relieved:

**Pull request merged**